chromium_src/components/payments/content/payment_request.cc

```diff
@@ -0,0 +1,123 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/payments/content/payment_request.h"
+
+#include "base/bind.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "chrome/browser/profiles/profile.h"
+#include "third_party/blink/public/mojom/payments/payment_request.mojom.h"
+
+using payments::mojom::PaymentErrorReason;
+
+#define Init Init_ChromiumImpl
+#include "../../../../../components/payments/content/payment_request.cc"
+#undef Init
+
+namespace payments {
+
+#if !defined(OS_ANDROID)
+
+void PaymentRequest::GetPublisherDetailsCallback(
+    mojo::PendingRemote<mojom::PaymentRequestClient> client,
+    std::vector<mojom::PaymentMethodDataPtr> method_data,
+    mojom::PaymentDetailsPtr details,
+    mojom::PaymentOptionsPtr options,
+    const ledger::type::Result result,
+    ledger::type::PublisherInfoPtr info) {
+  if (!info) {
+    log_.Error(brave_rewards::errors::kInvalidPublisher);
+    TerminateConnection();
+    return;
+  }
+
+  if (info->status == ledger::type::PublisherStatus::NOT_VERIFIED) {
+    log_.Error(brave_rewards::errors::kInvalidPublisher);
+    TerminateConnection();
+    return;
+  }
```

Comment:
Maybe we could combine these two checks to avoid duplicating the error handling?

Reply:
resolved